onnx/defs/tensor/defs.cc

```diff
@@ -89,6 +89,9 @@ the end of that dimension.
 reverse order.
 )DOC")
     .Input(0, "data", "Tensor of data to extract slices from.")
+    .Attr("axes",
```

Comment:
Just to make sure I understand this, so `axes=[1, 0]` means `data[starts[1]:ends[1], starts[0]:ends[0]]`? If so, since these are attributes, so I assume they are already known at the time frontends export onnx, why couldn't frontends reshuffle starts and ends?

Reply:
This isn't for shuffle, actually the above won't work. Let's say you have 5D tensor as input, and you want to Slice the 2 in the middle, you can say: axes = (2,3), start=(1,1), end=(4,4). So axes, specify the list of axes in which you apply the start and end indices, len(axes) = len(starts) = len(ends). This is better, then specifying all the axes in starts and ends, and most of the starts and ends are the entire dimension.

Reply:
I see that makes senses

Reply:
Another way is to allow -1 to denote "the rest" like in Python and always specify the sizes. TF does the same: https://www.tensorflow.org/api_docs/python/tf/slice

Same with C2: https://github.com/caffe2/caffe2/blob/01827c153db96349745a544148c1ff0386c5ef9e/caffe2/operators/slice_op.h#L59

I feel it's more natural. What do you think?

Reply:
-1 mean read from the end.

Reply:
As an fyi, in c2, slice don't have "axis" and user needs to specify the starts and ends of each dimension, and currently c2 only supports slicing one of the dimensions. We might need to change c2 implementation when this is merged.

Reply:
Yeah, I mean that instead of `axis` you can fill other dimensions with [0, -1]

In other words for 5D tensor and axes = (2,3), start=(1,1), end=(4,4) it'd become
`start=(0,0,1,1,0), end=(-1,-1,4,4,-1)` and it's more explicit.

Reply:
Also in numpy/matlab we need to specify all the dimensions in array slicing. I think the definition of c2 slice op is probably borrowed from numpy.